### PR TITLE
techdebt: [CDS-85088]: remove ff PIE_STATIC_YAML_SCHEMA

### PR DESCRIPTION
**Pull request by abhinavrastogi-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by abhinavrastogi-harness** [comment]
trigger messagecheck

**Review by vikrant-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx [hunk 1]
```diff
@@ -60,7 +57,7 @@ export function PipelineSchemaContextProvider(
     queryParams: {
       ...commonQueryParams
     } as GetSchemaYamlQueryParams,
-    lazy: negateImplication(__DEV__, !!PIE_STATIC_YAML_SCHEMA)
+    lazy: !__DEV__
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx:60]: @vikrant-harness Do we need this __DEV__ condition now, I think this was added for openAPI migration fallback?

Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx:60]: yes we still need this as open api is not yet supported for local BE development

Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx:60]: For other open APIs we do not follow the same approach, right?

Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx:60]: we should ideally follow this but open-api is not widely adopted yet so no one specifically complained about it , pipeline page is pretty common use case so we added it here , we can check other places where we need to add. (Plus it was non-blocking as it is only used for recommendations)

Comment (abhinavrastogi-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx:60]: @vikrant-harness what do we mean by
> open api is not yet supported for local BE development

we do support calling open-api apis on local. We just don't support per-service proxy for local development.

Comment (abhinavrastogi-harness) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext.tsx:60]: if needed, we can make an entry specific to this endpoint in devServerProxyConfig.js to unblock per-endpoint proxy as well.

**Pull request merged**